src/app-layer-parser.c

```diff
@@ -140,6 +140,8 @@ struct AppLayerParserState_ {
      * we don't need a var per direction since we don't log a transaction
      * unless we have the entire transaction. */
     uint64_t log_id;
+    /* Count of number of transactions */
+    uint64_t tx_cnt;
```

Comment:
I'm not convinced this is needed if it's only used to get a diff between tx_cnt before and after a AppLayerParse call.